go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -484,6 +484,15 @@ func createNodeManagementPorts(name string, nodeAnnotator kube.Annotator, waiter
 	return mgmtPorts, mgmtPortConfig, nil
 }
 
+func getOVNSBZone() string {
+	dbZone, _, err := util.RunOVNSbctl("get", "SB_Global", ".", "options:name")
```

Comment:
Its possible for the deployment to not set any zone name i,e options:name is not set.

This is how the zone is expected to be set by deployment user

- ovn-nbctl set NB_Global . name=<zone_name>
- ovn-nbctl set NB_Global . options:name=<zone_name>   ->   ovn-northd will copy the NB_Global.options column values to SB_Global.options.

That's how ovnkube-node would get the zone name.

In case the user has not set any zone name,  then I think its fair to assume the zone name is types.OvnDefaultZone.


So which means we need to handle the scenario when ovn-sbctl command can't  talk to the Southbound database.  Let me check how we can figure this out.

Reply:
Maybe the function runOVNretry function [here](https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L416)  should not return err if the command fails ? And instead the caller can rely on stderr ?  Or give a proper indication what is the error type ?  Error due to failure to connect to the dB or error due to command execution failed ?

Reply:
Ok.  Will this do ?

```
// getOVNSBZone returns the zone name stored in the Southbound db.
// It returns the default zone name if "options:name" is not set in the SB_Global row
func getOVNSBZone() (string, error) {
	dbZone, stderr, err := util.RunOVNSbctl("get", "SB_Global", ".", "options:name")
	if err != nil {
		if strings.Contains(stderr, "Connection refused") {
			// Some issue with the connection to the Southbound db server
			return "", err
		}
		// If the options:name is not present, assume default zone
		return types.OvnDefaultZone, nil
	}

	return dbZone, nil
}
```

Reply:
I definitely like that better than not distinguishing the 2 errors at all. Thank you so much for spending time to improve this error handling, @numansiddique !

Reply:
Not sure if you think that is better, but you could instead return global if `if strings.Contains(stderr, "ovn-sbctl: no key \"name\" in SB_Global record")` wdyt?